library/src/scala/runtime/stdLibPatches/Predef.scala

```diff
@@ -62,4 +62,8 @@ object Predef:
      *  `eq` or `ne` methods, only `==` and `!=` inherited from `Any`. */
     inline def ne(inline y: AnyRef | Null): Boolean =
       !(x eq y)
+
+  extension (inline opt: Option.type)
+    @experimental
+    inline def fromNullable[T](t: T | Null): Option[T] = Option(t).asInstanceOf[Option[T]]
```

Comment:
if you explicitly pass `Null` in the type it isn't eliminated:
```scala
scala> Option.fromNullable(1 : Int | Null)
val res10: Option[Int] = Some(1)

scala> Option.fromNullable[Int | Null](1 : Int | Null)
val res11: Option[Int | Null] = Some(1)
```

Reply:
This is expected behaviour, similar to `.nn`. There is not type-level "subtraction" to strip `Null`, so we can only rely on type inference to remove `Null` .

Reply:
makes sense